examples/self-hosted-databases/redis-cluster/docker-compose.yml

```diff
@@ -0,0 +1,85 @@
+services:
+  redis-node-0:
+    image: redis:7.0
+    volumes:
+      - "./:/opt/redis"
+    command: ["redis-server", "/opt/redis/redis.conf", "--tls-port", "17000"]
+    ports:
+      - '17000:17000'
+      - '17001:17001'
+      - '17002:17002'
+      - '17003:17003'
+      - '17004:17004'
+      - '17005:17005'
+
+  redis-node-1:
+    image: redis:7.0
+    volumes:
+      - "./:/opt/redis"
+    command: ["redis-server", "/opt/redis/redis.conf", "--tls-port", "17001", "--port", "0"]
+    network_mode: "service:redis-node-0"
```

Comment:
does `network_mode: "service:redis-node-0"` make it so that each node's listener is available on 127.0.0.1:{port} ?

Reply:
ya, it seems like what it is doing. the [doc](https://docs.docker.com/compose/compose-file/05-services/#network_mode) isn't clear what it is doing. I copied this hack from someone else doing a Redis cluster. I initially have the docker-compose assign each a local IP. But bridge mode does not work on MacOS so you can't access that network directly from host without port forwarding, so I tried this hack.